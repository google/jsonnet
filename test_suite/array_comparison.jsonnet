local arrs = [
  [],
  [1, 2],
  [1, 2, 3],
  [2, 3],
];
[
  {
    ['%s < %s' % [x, y]]: x < y,
    ['%s > %s' % [x, y]]: x > y,
    ['%s <= %s' % [x, y]]: x <= y,
    ['%s >= %s' % [x, y]]: x >= y,
  }
  for x in arrs
  for y in arrs
]
