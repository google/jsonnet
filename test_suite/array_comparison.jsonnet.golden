[
   {
      "[ ] < [ ]": false,
      "[ ] <= [ ]": true,
      "[ ] > [ ]": false,
      "[ ] >= [ ]": true
   },
   {
      "[ ] < [1, 2]": true,
      "[ ] <= [1, 2]": true,
      "[ ] > [1, 2]": false,
      "[ ] >= [1, 2]": false
   },
   {
      "[ ] < [1, 2, 3]": true,
      "[ ] <= [1, 2, 3]": true,
      "[ ] > [1, 2, 3]": false,
      "[ ] >= [1, 2, 3]": false
   },
   {
      "[ ] < [2, 3]": true,
      "[ ] <= [2, 3]": true,
      "[ ] > [2, 3]": false,
      "[ ] >= [2, 3]": false
   },
   {
      "[1, 2] < [ ]": false,
      "[1, 2] <= [ ]": false,
      "[1, 2] > [ ]": true,
      "[1, 2] >= [ ]": true
   },
   {
      "[1, 2] < [1, 2]": false,
      "[1, 2] <= [1, 2]": true,
      "[1, 2] > [1, 2]": false,
      "[1, 2] >= [1, 2]": true
   },
   {
      "[1, 2] < [1, 2, 3]": true,
      "[1, 2] <= [1, 2, 3]": true,
      "[1, 2] > [1, 2, 3]": false,
      "[1, 2] >= [1, 2, 3]": false
   },
   {
      "[1, 2] < [2, 3]": true,
      "[1, 2] <= [2, 3]": true,
      "[1, 2] > [2, 3]": false,
      "[1, 2] >= [2, 3]": false
   },
   {
      "[1, 2, 3] < [ ]": false,
      "[1, 2, 3] <= [ ]": false,
      "[1, 2, 3] > [ ]": true,
      "[1, 2, 3] >= [ ]": true
   },
   {
      "[1, 2, 3] < [1, 2]": false,
      "[1, 2, 3] <= [1, 2]": false,
      "[1, 2, 3] > [1, 2]": true,
      "[1, 2, 3] >= [1, 2]": true
   },
   {
      "[1, 2, 3] < [1, 2, 3]": false,
      "[1, 2, 3] <= [1, 2, 3]": true,
      "[1, 2, 3] > [1, 2, 3]": false,
      "[1, 2, 3] >= [1, 2, 3]": true
   },
   {
      "[1, 2, 3] < [2, 3]": true,
      "[1, 2, 3] <= [2, 3]": true,
      "[1, 2, 3] > [2, 3]": false,
      "[1, 2, 3] >= [2, 3]": false
   },
   {
      "[2, 3] < [ ]": false,
      "[2, 3] <= [ ]": false,
      "[2, 3] > [ ]": true,
      "[2, 3] >= [ ]": true
   },
   {
      "[2, 3] < [1, 2]": false,
      "[2, 3] <= [1, 2]": false,
      "[2, 3] > [1, 2]": true,
      "[2, 3] >= [1, 2]": true
   },
   {
      "[2, 3] < [1, 2, 3]": false,
      "[2, 3] <= [1, 2, 3]": false,
      "[2, 3] > [1, 2, 3]": true,
      "[2, 3] >= [1, 2, 3]": true
   },
   {
      "[2, 3] < [2, 3]": false,
      "[2, 3] <= [2, 3]": true,
      "[2, 3] > [2, 3]": false,
      "[2, 3] >= [2, 3]": true
   }
]
